pkg/controllerutils/finalizers.go

```diff
@@ -30,6 +30,28 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// PatchFinalizers adds the given finalizers to the object via a patch request.
+func PatchFinalizers(ctx context.Context, c client.Client, obj kutil.Object, finalizers ...string) error {
+	beforePatch := obj.DeepCopyObject()
+
+	f := sets.NewString(obj.GetFinalizers()...)
+	f.Insert(finalizers...)
+	obj.SetFinalizers(f.UnsortedList())
```

Comment:
Why is the order relevant?

Reply:
I think, `json-patch`es on arrays work with indices. That means, that switching the order of the finalizers, might lead to patches that don't result in the desired operation.

Reply:
I double checked and actually I think we can use patch safely in combination with `client.MergeFromWithOptimisticLock{}` because the element order during runtime is safe with `f.UnsortedList()`. WDYT? Falling back to update calls is still an option, I still don't want to miss a chance to pave the path towards patch request.

Reply:
I'm also still in favor of using Patch requests here. 
Why do you think, the element order is safe at runtime? I would expect, that when inserting the slice elements into a map (like in `sets.NewString`) and then range over it to insert them into a slice again (like in `Set.UnsortedList`) the order will not always be the same (as maps are unsorted).
So I guess, the safer option wrt to order would be doing something like in the c-r helper function (`controllerutil.AddFinalizer`).

Reply:
Okay makes sense. Thanks for explaining.